### DISCUSSION TITLE
Error: could not initialize proxy [com.TableName#2] - no session

### DISCUSSION THREAD

**Discussion by Shardul555** [graphql-java/graphql-java#3702] [Q&A] [answered]
I am trying to implement *Operation Caching* in graphql, but when I am using this piece of code, I am encountering error
``` could not initialize proxy [com.TableName#2] - no session ```
I am getting a few records, but for some of them, where I am trying to query heirarchically, this error is coming. 
May be it has to do with Asnychronous nature of cache provided, but not sure.
Please help how to change ``` public GraphQlSourceBuilderCustomizer sourceBuilderCustomizer() ``` method to resolve this error.
``` 
private final AsyncCache<String, PreparsedDocumentEntry> cache = Caffeine
        .newBuilder()
        .maximumSize(250)
        .expireAfterAccess(Duration.ofMinutes(20))
        .buildAsync();

PreparsedDocumentProvider preparsedDocumentProvider = new PreparsedDocumentProvider() {
    @Override
    public CompletableFuture<PreparsedDocumentEntry> getDocumentAsync(ExecutionInput executionInput, Function<ExecutionInput, PreparsedDocumentEntry> computeFunction) {
            return cache.get(
        executionInput.getQuery(),
        s -> computeFunction.apply(executionInput)
    );
    }
};

@Bean
    public GraphQlSourceBuilderCustomizer sourceBuilderCustomizer() {
      return (builder) ->
          builder.configureGraphQl(graphQlBuilder ->
              graphQlBuilder.preparsedDocumentProvider(preparsedDocumentProvider));
    }
      ```

**Comment by bbakerman** [comment, marked as answer]
Can you post a stack trace

> could not initialize proxy [com.TableName#2] - no session

This does not feel like anything to do with Spring or graphql-java

At a guess it sounds like Hibernate

https://www.baeldung.com/hibernate-initialize-proxy-exception

**Reply by Shardul555** [reply]
Thanks @bbakerman for your response, yes may be this is related to hibernate exception, though I was able to remove the error just by use of ``` Cache<String, PreparsedDocumentEntry> cache``` instead of Async Cache, so I will be closing this thread for now, will inform with more details once it is a bit clear for me as well.
